src/Operation/RemoteMergeOperation.php

```diff
@@ -171,7 +190,31 @@ private function createSourceBranch()
         }
 
         if ($this->squash) {
-            $this->gitHelper->squashCommits($this->targetBase, $sourceBranch, $this->forceSquash);
+            $this->gitHelper->squashCommits($this->targetBase, $sourceBranch, $this->forceSquash, $this->guardSync);
+        }
+
+        $currentBaseHeadCommit = $this->processHelper->runCommand(['git', 'rev-parse', $this->targetBase]);
+        $lastKnownCommonCommit = $this->processHelper->runCommand(['git', 'merge-base', '--fork-point', $this->targetBase, $sourceBranch]);
+
+        if ($currentBaseHeadCommit !== $lastKnownCommonCommit) {
+            if ($this->rebase) {
+                try {
+                    $this->processHelper->runCommand(['git', 'pull', '--rebase', $this->targetBase]);
+                } catch (\Exception $e) {
+                    // Error, abort the rebase operation
+                    $this->processHelper->runCommand(['git', 'rebase', '--abort'], true);
+
+                    throw new MergeWorkflowException(sprintf('Git rebase failed while trying to synchronize history against "%s".', $this->targetBase), 0, $e);
+                }
+
+                // Retrieve the commits again
+                $currentBaseHeadCommit = $this->processHelper->runCommand(['git', 'rev-parse', $this->targetBase]);
+                $lastKnownCommonCommit = $this->processHelper->runCommand(['git', 'merge-base', '--fork-point', $this->targetBase, $sourceBranch]);
```

Comment:
Since you removed the code that disables the `guardSync`, it makes it necessary to get these commits again. Essentially you only had to make one of the changes, either remove the code that disables the `guardSync` when rebase` is set, or remove the getting of the commits during rebase, which wouldn't have had any effect in that context. But by removing both, can lead to issues when both the `--ensure-sync` and `--rebase` flags is passed